src/components/Button/Button.tsx

```diff
@@ -1,33 +1,50 @@
-import React from "react";
+import React, { cloneElement, ElementType, isValidElement } from "react";
 import getExternalLinkProps from "../../util/getExternalLinkProps";
 import StyledButton from "./StyledButton";
-import { ButtonProps, variants, sizes } from "./types";
+import { ButtonProps, scales, variants } from "./types";
 
-const Button: React.FC<ButtonProps> = ({ startIcon, endIcon, children, external, isLoading, disabled, ...props }) => {
+const Button = <E extends ElementType = "button">(props: ButtonProps<E>): JSX.Element => {
+  const { startIcon, endIcon, external, className, isLoading, disabled, children, ...rest } = props;
   const internalProps = external ? getExternalLinkProps() : {};
   const isDisabled = isLoading || disabled;
+  const classNames = className ? [className] : [];
+
+  if (isLoading) {
+    classNames.push("pancake-button--loading");
+  }
+
+  if (isDisabled && !isLoading) {
+    classNames.push("pancake-button--disabled");
+  }
```

Comment:
Added BEM classnames in case consumer needs to target the loading and disabled state.